test/integration/selenium-helpers.js

```diff
@@ -146,23 +250,30 @@ class SeleniumHelper {
     /**
      * Find an element by xpath.
      * @param {string} xpath The xpath to search for.
-     * @param {string} timeoutMessage The message to use if the element is not found.
-     * A different (non-configurable) message will be used if the element is found but not visible.
      * @returns {Promise<webdriver.WebElement>} A promise that resolves to the element.
      */
-    async findByXpath (xpath, timeoutMessage = `findByXpath timed out for path: ${xpath}`) {
-        const el =
-            await this.driver.wait(until.elementLocated(By.xpath(xpath)), DEFAULT_TIMEOUT_MILLISECONDS, timeoutMessage);
-        await this.driver.wait(el.isDisplayed(), DEFAULT_TIMEOUT_MILLISECONDS, `${xpath} is not visible`);
-        return el;
+    async findByXpath (xpath) {
+        const outerError = new SeleniumHelperError('findByXpath failed', [{xpath}]);
+        try {
+            const el = await this.driver.wait(until.elementLocated(By.xpath(xpath)), DEFAULT_TIMEOUT_MILLISECONDS);
+            await this.driver.wait(el.isDisplayed(), DEFAULT_TIMEOUT_MILLISECONDS);
+            return el;
+        } catch (cause) {
+            throw await outerError.chain(cause, this.driver);
```

Comment:
It makes me sad to `await` here, but that's necessary if we're going to collect any information like the URL, logs, or page source from the web driver.

If we get here then `cause` will either be:
- a `SeleniumHelperError` that should already have tried to collect driver context, so this call to `chain` won't try again.
- some other kind of `Error`, the "leaf" of the error chain -- most likely a timeout or stale reference error -- in which case `chain` will attempt to add context info from the driver.

In either case, the `message` from `cause` will be appended to the message from `outerError` so the problem can be traced through all levels of the stack.